### PR TITLE
Add an Apply button to the Web UI when running as a Win32 Service

### PR DESCRIPTION
**Pull request by cgutman**
## Description
This PR adds an "Apply" button to the web UI on systems where Sunshine is running as a service. When clicked, it saves the settings and sends a SIGINT to itself to gracefully terminate. SunshineSvc will then restart it, which will load the new settings.

The "Apply" button only appears on Windows systems where Sunshine is running as a service, but the scaffolding is there to support this on Linux and Mac too (probably by just execing ourselves again).

@TheElixZammuto should probably review the config.html changes for correctness. I'm definitely no web development expert.

### Screenshot
image

Save button success text (when Apply button is available):
image

Apply button success text:
image

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

### CODE REVIEW

File: src_assets/common/assets/web/config.html [hunk 1]
```diff
@@ -765,11 +765,18 @@ <h1 class="my-4">Configuration</h1>
...
+  </div>
+  <div class="alert alert-success my-4" v-if="saved && !restart_supported">
+    <b>Success!</b> Restart Sunshine to apply changes.
+  </div>
+  <div class="alert alert-success my-4" v-if="restarted">
+    <b>Success!</b> Sunshine is restarting to apply changes.
   </div>
   <div class="mb-3 buttons">
     <button class="btn btn-primary" @click="save">Save</button>
+    <button class="btn btn-success" @click="apply" v-if="restart_supported">Apply</button>
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/config.html:779]: Maybe only show button when settings are saved?

`v-if="saved && restart_supported"`

And maybe when not restarted?

``v-if="saved && restart_supported && !restarted"``

File: src_assets/common/assets/web/config.html [hunk 2]
```diff
@@ -906,12 +916,32 @@ <h1 class="my-4">Configuration</h1>
...
+      apply() {
+        this.saved = this.restarted = false;
+        this.serialize();
         fetch("/api/config", {
           method: "POST",
           body: JSON.stringify(this.config),
         }).then((r) => {
-          if (r.status == 200) this.success = true;
+          if (r.status == 200) {
+            fetch("/api/restart", {
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/config.html:938]: Would be good to have this function on the troubleshooting page as well, with a restart button.

Comment (cgutman) [reply src_assets/common/assets/web/config.html:938]: Added!

image

File: src_assets/common/assets/web/config.html [hunk 3]
```diff
@@ -765,11 +765,18 @@ <h1 class="my-4">Configuration</h1>
       </div>
     </div>
   </div>
-  <div class="alert alert-success my-4" v-if="success">
-    <b>Success!</b> Restart Sunshine to apply changes
+  <div class="alert alert-success my-4" v-if="saved && restart_supported">
```
Comment (TheElixZammuto) [inline src_assets/common/assets/web/config.html:768]: nitpick: Probably a bit reduntant to make 3 duplicated divs for the alert. You can use the <span> tag to wrap the text and then use v-if on the span themselves.

**Review by TheElixZammuto — APPROVED** [review]

**Pull request merged**